pkg/volume/csi/nodeinfomanager/nodeinfomanager.go

```diff
@@ -113,21 +114,17 @@ func (nim *nodeInfoManager) InstallCSIDriver(driverName string, driverNodeID str
 		updateNodeIDInNode(driverName, driverNodeID),
 	}
 
-	if utilfeature.DefaultFeatureGate.Enabled(features.CSINodeInfo) {
+	if csiNodeInfoEnabled {
 		nodeUpdateFuncs = append(nodeUpdateFuncs, updateTopologyLabels(topology))
 	}
 
-	if utilfeature.DefaultFeatureGate.Enabled(features.AttachVolumeLimit) {
```

Comment:
According to the [KEP](https://github.com/kubernetes/enhancements/blob/93a90a4f5d255eb60d5c80cca30fc20d13e0ef5d/keps/sig-storage/20190408-volume-scheduling-limits.md#design-details), we should keep using this feature gate for this functionality:

```
Existing feature gate AttachVolumeLimit will be re-used for implementation of this KEP. The feature is already beta and is enabled by default.
```
In fact, the predicate still uses it: 

https://github.com/kubernetes/kubernetes/blob/4db3fccd6f1c5ae8deb687985a2c8be76e4d4ab5/pkg/scheduler/algorithm/predicates/csi_volume_predicate.go#L54-L57

Reply:
The KEP also [mentions](https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/20190408-volume-scheduling-limits.md#implementation-detail-for-all-csi-drivers) that the Kubelet should stop adding the limits to the `Node` object:

```
Scheduler will respect Node.status.allocatable and Node.status.capacity for CSI volumes if CSINode object is not available or has missing entry in CSINode.spec.drivers[xyz].allocatable during a deprecation period but kubelet will stop populating Node.status.allocatable and Node.status.capacity for CSI volumes. 
```

I have the impression that the KEP's idea was to do something similar to what we've done with the `CSINodeInfo` object/gate: we migrated it to beta and changed the underlying implementation. `AttachVolumeLimits` is already beta though, so I don't know if we can do the same here.

Thoughts?

Reply:
I think having kubelet stop reporting limits through Node object is fine as long as the scheduler still supports both for up to 2 version skew.

Reply:
> +1 I'm not sure if Node object gets deleted/recreated during node upgrade, but may be good to keep cleanup logic just in case.

Added the cleanup back

Reply:
To summarize: kubelet stop reporting limits through Node object, but cleanup logic is still there.